### DISCUSSION TITLE
No Bluetooth or Wifi connection

### DISCUSSION THREAD

**Discussion by Robertarduino** [ClassicDIY/SkyeTracker#33] [Q&A] [answered]
Hello,  I  try from several days to start system. First problem appears with esp32 module base at China module. My computer refuses connection; ). So I bought dedicated at CP chip version. Flashing is done w/o error. I put WIFI network new name and password.  But WiFi is not active or Bluetooth also. I can't connect to esp32 module by sky app. Can you advice what can be wrong ? Thank you.

**Comment by graham22** [comment]
I would suggest trying one of the ESP32 bluetooth / Wifi tutorial examples, ex: https://randomnerdtutorials.com/ to make sure the radio in the ESP32 chip is working.
The SkyeTracker code was written 2 years ago so it may not work with the newer ESP32 versions, for example the new ESP32-S2 no longer has bluetooth support.

**Comment by Robertarduino** [comment]
Finally I succeed by using ESP flasher tool. Question about MQTT server. its use for internal system needs or it can be connected with external server as well? Also question if its possible to use simple switches instead PWM?

**Reply by graham22** [reply]
MQTT is optional and is not required by the internal system.
The signal to the actuators are not PWM, the code will activate the actuator for the required duration based on the specified rate and length of the actuator.

**Reply by Robertarduino** [reply]
Using MQTT server its possible to update actuator length? in app its already defined but if I have different one it will be difficult.. Or set up by max time open/close if its possible for actuators

**Reply by graham22** [reply]
No, it just supports mode and moveTo (IOT.cpp::123)

**Reply by Robertarduino** [reply]
Question about speed of cylinder setup procedure ? It's expiremetal about min to max reaching? How to setup, to not lost accuracy for positioning?

**Reply by graham22** [reply]
The min/max depends on how your frame is built, for horizontal, set the panels to direct south (or North in southern hemisphere) when actuator if half extended, then, using a compass, measure the east & west azimuth when actuator is fully retracted and extended.
The code will calculate the amount of time to run the actuator for each 5 degree increment from the resulting sweep from east to west

**Comment by Robertarduino** [comment, marked as answer]
Finally I succed to start system.  Two issue observed. 1) When wifi activated sometimes its lost current time or change for different time zone. 
2) when it's end of day panel's are setup in the same elevation like sun and its going according sun elevation.  Suspected that I made wrong wiring.

**Reply by Robertarduino** [reply]
Wrong conetion of actuator

**Comment by Robertarduino** [comment]
To summerise:
Min elevation its 0dgr. So its flat position we can force this situation by pushing top arrow in app.
MAX elevation. depending of system structure not more then 90dgr. we can also force system by pushing down arrow in app.
East and West simple by measuring according compass indicator and our system structure.
After observation behavior of system. During initialization its going max east plus elevation maks position. So panel are in for ex 60dgr and 100 East. During day it should follow sun from east to west changing also elevation up to 30dgr. When sun is going to sleep :) elevation of panels should go to maks down. When its dark system return to maks East and 45 dgr of elevation.
If wind will be too strong system will move to 180dgr and 0 for elevation. 
Correct if Im in line with system behavior :)
